src/Api/Controllers/AccountsController.cs

```diff
@@ -370,6 +375,45 @@ public async Task PostKdf([FromBody] KdfRequestModel model)
         throw new BadRequestException(ModelState);
     }
 
+    [HttpPost("rotate-key")]
+    [RequireFeature(FeatureFlagKeys.KeyRotationImprovements)]
+    [Obsolete("Intended for future key rotation enhancements. Do not use.")]
+    public async Task RotateKey([FromBody] RotateUserKeyRequestModel model)
+    {
+        var user = await _userService.GetUserByPrincipalAsync(User);
+        if (user == null)
+        {
+            throw new UnauthorizedAccessException();
+        }
+
+        var dataModel = new RotateUserKeyData
+        {
+            // MasterPasswordHash = model.MasterPasswordHash,
+            // Key = model.Key,
+            // PrivateKey = model.PrivateKey,
+            // Ciphers = await _cipherValidator.ValidateAsync(user, model.Ciphers),
+            // Folders = await _folderValidator.ValidateAsync(user, model.Folders),
+            // Sends = await _sendValidator.ValidateAsync(user, model.Sends),
+            // EmergencyAccessKeys = await _emergencyAccessValidator.ValidateAsync(user, model.EmergencyAccessKeys),
+            // AccountRecoveryKeys = await _accountRecoveryValidator.ValidateAsync(user, model.AccountRecoveryKeys),
+        };
+
+        var result = await _rotateUserKeyCommand.RotateUserKeyAsync(dataModel);
+
+        if (result.Succeeded)
+        {
+            return;
+        }
+
+        foreach (var error in result.Errors)
+        {
+            ModelState.AddModelError(string.Empty, error.Description);
+        }
+
+        await Task.Delay(2000);
```

Comment:
❓ Why is there a delay here?

Reply:
I believe its to prevent timing attacks for the MP verification. I copied it from previous endpoint, but I've since consolidated the two endpoints.